keyboards/aplusx_pulsar/xboard_qs/keymaps/via/keymap.c

```diff
@@ -0,0 +1,371 @@
+        KC_TAB,  KC_Q,  KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,   KC_LBRC, KC_RBRC, KC_BSLS,           KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,  KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,   KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,        KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT, KC_SLSH,          KC_RSFT,                     KC_UP,
+        KC_LCTL, KC_LALT, KC_LGUI,                   KC_SPC,                       KC_RGUI,  MO(MAC_FN), KC_RALT, KC_RCTL,           KC_LEFT, KC_DOWN, KC_RGHT),
+
+    [MAC_FN]   = LAYOUT(/* Layer 3 ; MAC Function Layer */
+                                                                                                                                KC_PC1, KC_PC2, RGB_MOD,
+        INIT_3S,      KC_F1,   KC_F2,   KC_F3,   KC_F4,     KC_F5,   KC_F6,   KC_F7,   KC_F8,     KC_F9,   KC_F10,  KC_F11,  KC_F12, KC_TMODE, KC_F13, KC_F14, KC_F15,
+        DM_RSTP, DM_REC1, DM_REC2, DM_PLY1, DM_PLY2, _______, _______, _______, _______, _______, _______, NK_OFF,   NK_ON,  RGB_TOG,          RGB_SAI, RGB_HUI, RGB_MOD,
+        _______, KC_BTN1, KC_MS_U, KC_BTN2, KC_WH_U, _______, _______, _______, _______, _______, _______, _______, _______, _______,          RGB_SAD, RGB_HUD, RGB_RMOD,
+        _______, KC_MS_L, KC_MS_D, KC_MS_R, KC_WH_D, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,         KC_BTN4, KC_BTN3, KC_BTN5, _______, KC_S_LED, _______, _______, _______, _______, _______,          _______,                   RGB_VAI,
+        _______, _______, _______,                         _______,                               _______, _______, _______,  KC_TPC,          RGB_SPD, RGB_VAD, RGB_SPI)
+        };
+//-----------------------------------------------------------------
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [WIN_BASE] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [WIN_FN]   = { ENCODER_CCW_CW(RGB_VAD, RGB_VAI) },
+    [MAC_BASE] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [MAC_FN]   = { ENCODER_CCW_CW(RGB_VAD, RGB_VAI) }
+};
+#endif // ENCODER_MAP_ENABLE
+//-----------------------------------------------------------------
+#define HCS(report) host_consumer_send(record->event.pressed ? report : 0); return false
+#define HSS(report) host_system_send(record->event.pressed ? report : 0); return false
+
+static uint8_t mac_keycode[4] = {KC_LOPT, KC_ROPT, KC_LCMD, KC_RCMD};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+    switch (keycode) {
+        case INIT_3S:
+            if (record->event.pressed){
+                INIT3S_timer = timer_read();
+                INIT3S_on    = true;
+                return false;
+            }
+            else{
+                INIT3S_on   = false;
+                return false;
+            }
+            break;
```

Comment:
```suggestion
            }
            else{
                INIT3S_on   = false;
            }
            break;
```

Reply:
Thank you for your careful consideration